### ISSUE TITLE
Cannot use BuildImages AMAZON_LINUX_2_4 or STANDARD_6_0 in deployment_maps 

### ISSUE THREAD

**Issue by mirceaadam** [awslabs/aws-deployment-framework#503]
Hi,

In our organization we have started to upgrade our lambdas and other projects to nodejs16.x and we have a couple of ADF CodeBuild projects which use ubuntu "STANDARD_6_0" or even "AMAZON_LINUX_2_4". 

I noticed that our ADF master pipeline is using cdk@1.88 and the newer cdk@2.39.0 (for example) has in the class LinuxBuildImage: STANDARD_6_0 and AMAZON_LINUX_2_4. 
Reference: 
https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_codebuild.LinuxBuildImage.html
https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-codebuild.LinuxBuildImage.html

At the moment if I need to update the BuildImage, I do it manually in the aws interface. 
I wonder, to do it via code as we normally do, is there a possibility to integrate this in the deployment_maps yml files ? Or do I need to upgrade ADF somehow ?

Here is also the error when trying to use AMAZON_LINUX_2_4 for example:
``` 

2022-08-26 07:56:02,180 \| INFO \| cdk_stacks.main \| Pipeline creation/update of my-FRONTEND-dev commenced \| (main.py:31)
--
334 | Traceback (most recent call last):
335 | File "adf-build/cdk/generate_pipeline_stacks.py", line 37, in <module>
336 | main()
337 | File "adf-build/cdk/generate_pipeline_stacks.py", line 33, in main
338 | PipelineStack(app, stack_input)
339 | File "/codebuild/output/src839325897/src/adf-build/jsii/_runtime.py", line 83, in __call__
340 | inst = super().__call__(*args, **kwargs)
341 | File "/codebuild/output/src839325897/src/adf-build/cdk/cdk_stacks/main.py", line 63, in __init__
342 | adf_codebuild.CodeBuild(
343 | File "/codebuild/output/src839325897/src/adf-build/jsii/_runtime.py", line 83, in __call__
344 | inst = super().__call__(*args, **kwargs)
345 | File "/codebuild/output/src839325897/src/adf-build/cdk/cdk_constructs/adf_codebuild.py", line 83, in __init__
346 | build_image=CodeBuild.determine_build_image(scope, target, map_params),
347 | File "/codebuild/output/src839325897/src/adf-build/cdk/cdk_constructs/adf_codebuild.py", line 186, in determine_build_image
348 | return getattr(
349 | AttributeError: type object 'LinuxBuildImage' has no attribute 'AMAZON_LINUX_2_4'
350 | Subprocess exited with error 1
351 |  
352 | [Container] 2022/08/26 07:56:02 Command did not exit successfully cdk synth --app adf-build/cdk/generate_pipeline_stacks.py 1> /dev/null exit status 1
353 | [Container] 2022/08/26 07:56:02 Phase complete: BUILD State: FAILED
```

Thank you,
Mircea

**Comment by sbkok** [comment]
Hi @mirceaadam,

You are on the right track.
To specify the CodeBuild image to use, you can use the `image` property of the CodeBuild provider in the deployment map.
The documentation on this property can be found here: https://github.com/awslabs/aws-deployment-framework/blob/master/docs/providers-guide.md#codebuild.

You can find an example here: https://github.com/awslabs/aws-deployment-framework/tree/master/samples/sample-serverless-app

As you discovered, v3.1.2 of ADF has not been updated recently.
We were blocked with another issue that had to be resolved first. We cleared that now and are working on the next release as we speak. The next version of ADF, v3.2.0, will update CDK to v1.169.

Updating to CDK v2 would require releasing a major version, as that would introduce breaking changes for any customization that was made to ADF.

I checked the NodeJS runtime support in CodeBuild, as you mentioned that requires STANDARD_6_0 or Amazon Linux 4, see https://docs.aws.amazon.com/codebuild/latest/userguide/available-runtimes.html.
While CDK v1.88 supports up to STANDARD_5_0, see https://github.com/aws/aws-cdk/blob/v1.88.0/packages/%40aws-cdk/aws-codebuild/lib/project.ts#L1480

So I am afraid you will have to await the next release to update the pipelines accordingly.
Or manually apply the patch that updates the CDK version as we did in the main branch of this repository.

I hope this helps you to move forward or at least gives you clarity on why and how it will be supported later on.

**Comment by mirceaadam** [comment]
Hi @sbkok , thank you for the fast reply - this helps, thank you!

**Comment by sbkok** [comment]
Thank you for your patience. I am happy to inform you that the update to a recent version of CDK has been [released as part of v3.2.0](https://github.com/awslabs/aws-deployment-framework/releases/tag/v3.2.0) just now.
Please open a new issue if you are experiencing any issues related to this still.

**Comment by Wildpipe** [comment]
This is still an issue in ADF 3.2.0. Picking STANDARD_6_0 as an image yields the following error:

`<html>
<body>


2023-03-10 09:22:06,207 \| INFO \| cdk_stacks.main \| Pipeline creation/update of eks-template commenced \| (main.py:36)
--
136 | Traceback (most recent call last):
137 | File "/codebuild/output/src157726161/src/adf-build/cdk/generate_pipeline_stacks.py", line 39, in <module>
138 | main()
139 | File "/codebuild/output/src157726161/src/adf-build/cdk/generate_pipeline_stacks.py", line 34, in main
140 | PipelineStack(app, stack_input)
141 | File "/codebuild/output/src157726161/src/adf-build/jsii/_runtime.py", line 112, in __call__
142 | inst = super().__call__(*args, **kwargs)
143 | File "/codebuild/output/src157726161/src/adf-build/cdk/cdk_stacks/main.py", line 47, in __init__
144 | self.generate_pipeline(pipeline_type, stack_input)
145 | File "/codebuild/output/src157726161/src/adf-build/cdk/cdk_stacks/main.py", line 51, in generate_pipeline
146 | generate_default_pipeline(self, stack_input)
147 | File "/codebuild/output/src157726161/src/adf-build/cdk/cdk_stacks/adf_default_pipeline.py", line 56, in generate_adf_default_pipeline
148 | build_stage = _generate_build_stage_for_pipeline(scope, stack_input)
149 | File "/codebuild/output/src157726161/src/adf-build/cdk/cdk_stacks/adf_default_pipeline.py", line 158, in _generate_build_stage_for_pipeline
150 | return adf_codebuild.CodeBuild(
151 | File "/codebuild/output/src157726161/src/adf-build/jsii/_runtime.py", line 112, in __call__
152 | inst = super().__call__(*args, **kwargs)
153 | File "/codebuild/output/src157726161/src/adf-build/cdk/cdk_constructs/adf_codebuild.py", line 165, in __init__
154 | build_image=CodeBuild.determine_build_image(
155 | File "/codebuild/output/src157726161/src/adf-build/cdk/cdk_constructs/adf_codebuild.py", line 385, in determine_build_image
156 | return CodeBuild.get_image_by_name(specific_image)
157 | File "/codebuild/output/src157726161/src/adf-build/cdk/cdk_constructs/adf_codebuild.py", line 353, in get_image_by_name
158 | raise Exception(
159 | Exception: The CodeBuild image STANDARD_6_0 could not be found.


</body>
</html>`
